applications/FluidDynamicsApplication/custom_processes/shock_capturing_process.cpp

```diff
@@ -0,0 +1,442 @@
+        return aux_TLS_container;
+    }
+
+    /**
+     * @brief Physics-based shock capturing
+     * This function calculates the artificial magnitudes using a physics-based shock capturing method.
+     * References https://arc.aiaa.org/doi/abs/10.2514/6.2018-0062
+     */
+    void ShockCapturingProcess::CalculatePhysicsBasedShockCapturing()
+    {
+        // Calculate the model part data
+        const auto r_process_info = mrModelPart.GetProcessInfo();
+        const int n_nodes = mrModelPart.NumberOfNodes();
+
+        // Initialize the values to zero
+        block_for_each(mrModelPart.Nodes(), [](Node<3> &rNode) {
+            rNode.GetValue(ARTIFICIAL_CONDUCTIVITY) = 0.0;
+            rNode.GetValue(ARTIFICIAL_BULK_VISCOSITY) = 0.0;
+            rNode.GetValue(ARTIFICIAL_DYNAMIC_VISCOSITY) = 0.0;
+        });
+
+        // If required, update the NODAL_AREA
+        if (mUpdateNodalArea) {
+            CalculateNodalAreaProcess<false> nodal_area_process(mrModelPart);
+            nodal_area_process.Execute();
+        }
+
+        // Calculate the elemental contributions of the shock capturing
+        const auto geometry_type = (mrModelPart.ElementsBegin()->GetGeometry()).GetGeometryType();
+        if (geometry_type == GeometryData::KratosGeometryType::Kratos_Triangle2D3) {
+            // Set auxiliary TLS container and elemental function
+            ShockCapturingTLSType2D3N tls_container_2D3N = SetTLSContainer2D3N();
+            auto aux_function_2D3N = [&, this] (Element &rElement, ShockCapturingTLSType2D3N &rShockCapturingTLS) {this->CalculatePhysicsBasedShockCapturingElementContribution<2,3>(rElement, rShockCapturingTLS);};
+            // Perform the elemental loop
+            block_for_each(mrModelPart.Elements(), tls_container_2D3N, aux_function_2D3N);
+        } else if (geometry_type == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4) {
+            // Set auxiliary TLS container and elemental function
+            ShockCapturingTLSType3D4N tls_container_3D4N = SetTLSContainer3D4N();
+            auto aux_function_3D4N = [&, this] (Element &rElement, ShockCapturingTLSType3D4N &rShockCapturingTLS) {this->CalculatePhysicsBasedShockCapturingElementContribution<3,4>(rElement, rShockCapturingTLS);};
+            // Perform the elemental loop
+            block_for_each(mrModelPart.Elements(), tls_container_3D4N, aux_function_3D4N);
+        } else {
+            KRATOS_ERROR << "Asking for a non-supported geometry. Physics-based shock capturing only supports \'Triangle2D3\' and \'Tetrahedra3D4\' geometries.";
+        }
+
+        // Nodal smoothing of the shock capturing magnitudes
+        // Note that to avoid calculating the NODAL_AREA we took from the first DOF of the lumped mass vector
+        IndexPartition<>(n_nodes).for_each([&](const ModelPart::SizeType iNode) {
```

Comment:
why not the block_for_each here?

Reply:
I think it is a leftover from the previous strategy implementation. I'll change it. Thanks for noticing!

Reply:
Done.